expr/func_removeabovebelowpercentile.go

```diff
@@ -0,0 +1,106 @@
+package expr
+
+import (
+	"errors"
+	"fmt"
+	"math"
+	"sort"
+
+	schema "gopkg.in/raintank/schema.v1"
+
+	"github.com/grafana/metrictank/api/models"
+)
+
+type FuncRemoveAboveBelowPercentile struct {
+	in    GraphiteFunc
+	n     float64
+	above bool
+}
+
+func NewRemoveAboveBelowPercentileConstructor(above bool) func() GraphiteFunc {
+	return func() GraphiteFunc {
+		return &FuncRemoveAboveBelowPercentile{above: above}
+	}
+}
+
+func (s *FuncRemoveAboveBelowPercentile) Signature() ([]Arg, []Arg) {
+
+	return []Arg{
+		ArgSeriesList{val: &s.in},
+		ArgFloat{key: "n", val: &s.n},
+	}, []Arg{ArgSeriesList{}}
+}
+
+func (s *FuncRemoveAboveBelowPercentile) Context(context Context) Context {
+	return context
+}
+
+func (s *FuncRemoveAboveBelowPercentile) Exec(cache map[Req][]models.Series) ([]models.Series, error) {
+	series, err := s.in.Exec(cache)
+	if err != nil {
+		return nil, err
+	}
+
+	if s.n <= 0 {
+		return nil, errors.New("The requested percent is required to be greater than 0")
+	}
```

Comment:
should be a validator on the arg as returned by signature. that way we can check it in advance, instead of waiting until we've done a bunch of other work like fetching the data.